kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -285,6 +285,17 @@ class BatchJobSubmission(
       }
 
       MetricsSystem.tracing(_.decCount(MetricRegistry.name(OPERATION_OPEN, opType)))
+
+      if (!builder.processLaunched) {
+        builder.close()
+        if (recoveryMetadata.isDefined) {
+          killMessage = killBatchApplication()
```

Comment:
what' the difference in recovery mode ? there can not exist application, no ?

Reply:
to keep the state align, we have to kill the running application for recovery mode if did exist.

Reply:
how can the application live if builder is not processLaunched ?

Reply:
before recovery:
- 1. the batch is in PENDING state and process is running
- 2. kyuubi server crash, but the process has submitted the app but the batch state is still PENDING
recovery:
- 1. the batch builder is not launched
- 2. but the batch application is running